plugins/quota/actor.go

```diff
@@ -0,0 +1,87 @@
+package quota
+
+import (
+	"os"
+	"path"
+
+	"github.com/gluster/glusterd2/glusterd2/daemon"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/glusterd2/xlator"
+	log "github.com/sirupsen/logrus"
+)
+
+type quotadActor struct{}
+
+const (
+	quotaDaemonKey = "enable"
+)
+
+// IsQuotadAffectedNode returns true if there are local bricks of volume on
+// which quota is enabled
+func isQuotadAffectedNode(logger log.FieldLogger) bool {
+	volumes, err := volume.GetVolumes()
+	if err != nil {
+		logger.WithError(err).Error("Failed to get volumes")
+		return true
+	}
+	for _, v := range volumes {
+		val, exists := v.Options[quotaEnabledKey]
+		if exists && val == "off" {
+			continue
+		} else if v.State != volume.VolStarted {
+			continue
+		} else {
+			bricks := v.GetLocalBricks()
+			if len(bricks) > 0 {
+				return true
+			}
+			continue
+		}
+	}
+	return false
+}
+
+func (actor *quotadActor) Do(v *volume.Volinfo, key string, value string, logger log.FieldLogger) error {
+	if key != quotaDaemonKey {
+		return nil
+	}
+	quotadDaemon, err := NewQuotad()
+	if err != nil {
+		return err
+	}
+	// Create pidfile dir if not exists
+	if err := os.MkdirAll(path.Dir(quotadDaemon.pidfilepath),
+		os.ModeDir|os.ModePerm); err != nil {
+		return err
+	}
+
+	// Create logFiledir dir
+	if err := os.MkdirAll(path.Dir(quotadDaemon.logfilepath),
+		os.ModeDir|os.ModePerm); err != nil {
+		return err
+	}
+
+	// Should the bitd and scrubber untouched on this node
+	if !isQuotadAffectedNode(logger) {
+		// This condition is for disable
+		// TODO: Need to ignore errors where process is already running.
+		daemon.Stop(quotadDaemon, true, logger)
+	} else {
+		//TODO: Handle ENOENT of pidFile
+		err = daemon.Stop(quotadDaemon, true, logger)
```

Comment:
The first error gets ignored. You should handle that or at lease log it.

Reply:
done